src/algebra/order/hom/ring.lean

```diff
@@ -150,6 +196,105 @@ lemma cancel_left {f : β →+*o γ} {g₁ g₂ : α →+*o β} (hf : injective
+  map_add := λ f, f.map_add',
+  map_mul := λ f, f.map_mul',
+  map_le_map_iff := λ f, f.map_le_map_iff',
+  left_inv := λ f, f.left_inv,
+  right_inv := λ f, f.right_inv }
+
+/-- Helper instance for when there's too many metavariables to apply `fun_like.has_coe_to_fun`
+directly. -/
+instance : has_coe_to_fun (α ≃+*o β) (λ _, α → β) := fun_like.has_coe_to_fun
+
+lemma to_fun_eq_coe (f : α ≃+*o β) : f.to_fun = f := rfl
+
+@[ext] lemma ext {f g : α ≃+*o β} (h : ∀ a, f a = g a) : f = g := fun_like.ext f g h
+
+@[simp] lemma coe_mk (e : α ≃+* β) (h) : ⇑(⟨e, h⟩ : α ≃+*o β) = e := rfl
+@[simp] lemma mk_coe (e : α ≃+*o β) (h) : (⟨e, h⟩ : α ≃+*o β) = e := ext $ λ _, rfl
+
+@[simp] lemma to_ring_equiv_eq_coe (f : α ≃+*o β) : f.to_ring_equiv = f := ring_equiv.ext $ λ _, rfl
+
+@[simp] lemma to_order_iso_eq_coe (f : α ≃+*o β) : f.to_order_iso = f := order_iso.ext $ λ _, rfl
+
+@[simp, norm_cast] lemma coe_to_ring_equiv (f : α ≃+*o β) : ⇑(f : α ≃+* β) = f := rfl
+@[simp, norm_cast] lemma coe_to_order_iso (f : α ≃+*o β) : ⇑(f : α ≃o β) = f := rfl
+
+variable (α)
+
+/-- The identity map as a ring isomorphism. -/
+@[refl] protected def refl : α ≃+*o α := ⟨ring_equiv.refl α, λ _ _, iff.rfl⟩
+
+instance : inhabited (α ≃+*o α) := ⟨order_ring_iso.refl α⟩
+
+@[simp] lemma refl_apply (x : α) : order_ring_iso.refl α x = x := rfl
+@[simp] lemma coe_ring_equiv_refl : (order_ring_iso.refl α : α ≃+* α) = ring_equiv.refl α := rfl
+@[simp] lemma coe_order_iso_refl : (order_ring_iso.refl α : α ≃o α) = order_iso.refl α := rfl
+
+variables {α}
+
+/-- The inverse of an ordered ring isomorphism as an ordered ring isomorphism. -/
+@[symm] protected def symm (e : α ≃+*o β) : β ≃+*o α :=
+⟨e.to_ring_equiv.symm,
+  λ a b, by erw [←map_le_map_iff e, e.1.apply_symm_apply, e.1.apply_symm_apply]⟩
+
+/-- See Note [custom simps projection] -/
+def simps.symm_apply (e : α ≃+*o β) : β → α := e.symm
+
+@[simp] lemma symm_symm (e : α ≃+*o β) : e.symm.symm = e := ext $ λ _, rfl
+
+/-- Composition of `order_ring_iso`s as an `order_ring_iso`. -/
+@[trans, simps] protected def trans (f : α ≃+*o β) (g : β ≃+*o γ) : α ≃+*o γ :=
+⟨f.to_ring_equiv.trans g.to_ring_equiv, λ a b, (map_le_map_iff g).trans (map_le_map_iff f)⟩
+
+@[simp] lemma trans_apply (f : α ≃+*o β) (g : β ≃+*o γ) (a : α) : f.trans g a = g (f a) := rfl
+
+@[simp] lemma self_trans_symm (e : α ≃+*o β) : e.trans e.symm = order_ring_iso.refl α :=
+ext e.left_inv
+@[simp] lemma symm_trans_self (e : α ≃+*o β) : e.symm.trans e = order_ring_iso.refl β :=
+ext e.right_inv
+
+end has_le
+
+section non_assoc_semiring
+variables [non_assoc_semiring α] [preorder α] [non_assoc_semiring β] [preorder β]
+  [non_assoc_semiring γ] [preorder γ]
+
+/-- Reinterpret an ordered ring isomorphism as an ordered ring homomorphism. -/
+def to_order_ring_hom (f : α ≃+*o β) : α →+*o β :=
+⟨f.to_ring_equiv.to_ring_hom, λ a b, (map_le_map_iff f).2⟩
+
+lemma to_order_ring_hom_eq_coe (f : α ≃+*o β) : f.to_order_ring_hom = f := rfl
```

Comment:
Do we want to be bold and choose a `simp` normal form?
```suggestion
@[simp]
lemma to_order_ring_hom_eq_coe (f : α ≃+*o β) : f.to_order_ring_hom = f := rfl
```

Reply:
Yes we do!